privaterelay/management/commands/aggregate_generated_for.py

```diff
@@ -0,0 +1,129 @@
+            "total_usage",  # Sum of emails forwarded, emails blocked,
+            # trackers blocked in emails, emails replied, and spam blocked
+            "total_forwarded",  # Total emails forwarded to masks
+            "total_blocked",  # Total emails blocked for masks
+            "total_level_one_trackers_blocked",  # Total number of trackers
+            # blocked in emails forwarded to masks
+            "total_replied",  # Total number of emails replied to masks
+            "total_spam",  # Total number of spam
+        ]
+
+        for row in datareader:
+            aggregate_data: dict[str, int] = {
+                "count": 0,
+                "row_count": 0,
+                "total_usage": 0,
+                "total_forwarded": 0,
+                "total_blocked": 0,
+                "total_level_one_trackers_blocked": 0,
+                "total_replied": 0,
+                "total_spam": 0,
+            }
+            url = row["generated_for"]
+
+            # TODO: clean the domain for multiple domains in generated_for
+            # separated by space, strip www, and others like stripping path
+            if url:
+                normalized_url = normalize(url)
+                domain = urlparse(normalized_url).netloc
+            else:
+                domain = url
+
+            if domain in aggregate_usage:
+                aggregate_data = aggregate_usage[domain]
+
+            aggregate_data["row_count"] = aggregate_data["row_count"] + 1
+            for col in columns:
+                d = 0
+                try:
+                    d = int(row[col])
+
+                except ValueError as e:
+                    print(e)
+                    print(f"{col}: {row[col]}")
+                aggregate_data[col] += d
+            aggregate_usage[domain] = aggregate_data
+
+    tmp_dir = Path(file_path).parent
+    aggregate_file_path = tmp_dir.joinpath("aggregate.csv")
+    with open(aggregate_file_path, "w", newline="") as csvfile:
+        field_names = [
+            "domain",
+            "rank",
+            "count",
+            "row_count",
+            "total_usage",
+            "ratio_usage",
+            "total_forwarded",
+            "ratio_forwarded",
+            "total_blocked",
+            "ratio_blocked",
+            "total_level_one_trackers_blocked",
+            "ratio_level_one_trackers_blocked",
+            "total_replied",
+            "ratio_replied",
+            "total_spam",
+            "ratio_spam",
+        ]
+        writer = DictWriter(csvfile, fieldnames=field_names)
+
+        writer.writeheader()
+        for k, v in aggregate_usage.items():
+            row = {"domain": k}
+            row.update(v)
+            writer.writerow(row)
+    return aggregate_file_path
+
+
+class Command(BaseCommand):
+    help = (
+        "Normalizes URLs in domain column and aggregates the values. "
+        "Creates or updates aggregate.csv."
+    )
```

Comment:
Nit: You might want to include a MPP reference so staff can track down how to create the `generated_for.csv` file

Reply:
Added MPP reference